lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2557,7 +2560,7 @@ HSADispatch::waitComplete() {
         printf("Signal wait returned unexpected value\n");
         exit(0);
     }
-
+    dispose();
```

Comment:
Hi Jack, by removing it, we can't reclaim the signals. I couldn't get what you want me to change.